### PR TITLE
Feat/stylelint set order of content #6035

### PR DESCRIPTION
**Pull request by chriddell**
Resolves https://github.com/wellcometrust/corporate/issues/6035

- order of content:
  - `@include` (blockless)
  - `@extend`
  - custom-properties
  - declarations
  - `@include` (with block)
  - pseudo-states
  - stateful rules (e.g. &.is-active, &.has-state)
  - `@include mq` (media queries)
  - pseudo-elements (before/after)
  - rules (nested rules)
- forces new line before at-rules
  - except first nested at-rule
  - except blockless after blockless
  - ignores at-rule after a comment

### CODE REVIEW

**Comment by chriddell** [comment]
Full stylelint options if you wanted to test different scenarios: https://raw.githubusercontent.com/wellcometrust/corporate-components/edbca7ed7c4f4600a66a95c0720addedc467ecb8/.stylelintrc.json

File: src/components/Hamburger/_hamburger.scss [hunk 1]
```diff
@@ -8,11 +8,11 @@
     color: var(--colour-white);
   }
 
-  &[disabled] {
+  @include mq(md) {
     display: none;
   }
 
-  @include mq(md) {
+  &[disabled] {
```
Comment (creido-welly) [inline src/components/Hamburger/_hamburger.scss:15]: I reckon this qualifies as a state so should be added to the rule for .is-\.has

Comment (chriddell) [reply src/components/Hamburger/_hamburger.scss:15]: 👍

File: .stylelintrc.json [hunk 2]
```diff
@@ -18,16 +18,57 @@
...
                     "hasBlock": false
-                }
+                },
+                {
+                    "type": "at-rule",
+                    "name": "extend"
+                },
+                "custom-properties",
+                "declarations",
+                {
```
Comment (creido-welly) [inline .stylelintrc.json:48]: presumably this would include @include hocus so should that go underneath mq?

Comment (chriddell) [reply .stylelintrc.json:48]: good point, I would say that should live with pseudo-states

**Review by creido-welly — APPROVED** [review]

**Pull request merged**